source/dxgi/dxgi_swapchain.cpp

```diff
@@ -297,6 +298,29 @@ HRESULT STDMETHODCALLTYPE DXGISwapChain::GetDevice(REFIID riid, void **ppDevice)
 
 HRESULT STDMETHODCALLTYPE DXGISwapChain::Present(UINT SyncInterval, UINT Flags)
 {
+#if RESHADE_ADDON
+	// Emit PRESENT_START latency marker before ReShade present handling
+	if (_is_initialized)
+	{
+		const uint64_t ts_ns = std::chrono::duration_cast<std::chrono::nanoseconds>(std::chrono::high_resolution_clock::now().time_since_epoch()).count();
+		switch (_direct3d_version)
+		{
+		case reshade::api::device_api::d3d10:
+		case reshade::api::device_api::d3d11:
+			reshade::invoke_addon_event<reshade::addon_event::latency_present_start>(
+				static_cast<D3D11Device *>(static_cast<ID3D11Device *>(_direct3d_device))->_immediate_context,
+				_impl,
+				ts_ns);
+			break;
+		case reshade::api::device_api::d3d12:
+			reshade::invoke_addon_event<reshade::addon_event::latency_present_start>(
+				static_cast<D3D12CommandQueue *>(_direct3d_command_queue),
+				_impl,
+				ts_ns);
```

Comment:
This add-on is doing the same thing as `reshade::addon_event::present`, so can also just use that instead of introducing an additional one.